Add event tracking

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Adds event tracking via Google Analytics. If Site Kit is installed and has analytics configured, every popup will report custom events:

1. Seen
1. Dismissed - by clicking on close button
1. Dismissed - by clicking on "not interested" text
1. Any link click in popup
1. Any form submission in popup

GA custom events have three properties: `Category`, `Action`, `Label`. This PR proposes the following use:

| Category | Action    | Label     |
| :------------- | :------------- | :------------- |
| `Newspack Popup` | `Popup (link clicked \| form submitted \| not interesting \| dismissed \| seen)`       | `Newspack Popup: <popup title>`       |

Closes #64.

### How to test the changes in this Pull Request:

1. Make sure Site Kit is connected to Analytics
1. Create a popup with a link and a form (e.g. Button block and Mailchimp block)
1. Visit site with a popup and open devtools network tab. Filter `google-analytics`.
1. Trigger and observe reported events being sent:
    1. Seen
    1. Dismissed - by clicking on close button
    1. Dismissed - by clicking on "not interested" text
    1. Any link click in popup
    1. Any form submission in popup
1. Visit your [GA dashboard](https://analytics.google.com/analytics/web), navigate to "Behavior"->"Events"->"Overview" in the sidebar menu
1. Observe that the events are inspectable there
1. Try with both inline and regular popup

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+			return '';
+		}
+
+		$event_category = 'Newspack Popup';
```
```suggestion
		$event_category = 'Newspack Announcement';
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+		}
+
+		$event_category = 'Newspack Popup';
+		$event_label    = 'Newspack Popup: ' . $popup['title'];
```
```suggestion
		$event_label    = 'Newspack Announcement: ' . $popup['title'];
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+							"on": "click",
+							"request": "event",
+							"vars": {
+								"event_name": "<?php echo esc_html__( 'Popup link clicked' ); ?>",
```
```suggestion
								"event_name": "<?php echo esc_html__( 'Link Click', 'newspack-popups' ); ?>",
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+								"request": "event",
+								"selector": "#<?php echo esc_attr( $element_id ); ?> form:not(.popup-action-form)",
+								"vars": {
+									"event_name": "<?php echo esc_html__( 'Popup form submitted' ); ?>",
```
```suggestion
									"event_name": "<?php echo esc_html__( 'Form Submission', 'newspack-popups' ); ?>",
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+								"request": "event",
+								"selector": "#<?php echo esc_attr( $element_id ); ?> form.popup-dismiss-form",
+								"vars": {
+									"event_name": "<?php echo esc_html__( 'Popup dismissed' ); ?>",
```
```suggestion
									"event_name": "<?php echo esc_html__( 'Dismissal', 'newspack-popups' ); ?>",
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+							"request": "event",
+							"selector": "#<?php echo esc_attr( $element_id ); ?> form.popup-not-interested-form",
+							"vars": {
+								"event_name": "<?php echo esc_html__( 'Popup not interesting' ); ?>",
```
```suggestion
								"event_name": "<?php echo esc_html__( 'Permanent Dismissal', 'newspack-popups' ); ?>",
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+							"request": "event",
+							"selector": "#<?php echo esc_attr( $element_id ); ?>",
+							"vars": {
+								"event_name": "<?php echo esc_html__( 'Popup seen' ); ?>",
```
```suggestion
								"event_name": "<?php echo esc_html__( 'Seen', 'newspack-popups' ); ?>",
```

jeffersonrabb commented on includes/class-newspack-popups-model.php:
```diff
@@ -279,6 +279,115 @@ protected static function create_popup_object( $post, $include_categories = fals
+		$event_category = 'Newspack Popup';
+		$event_label    = 'Newspack Popup: ' . $popup['title'];
+
+		$has_link                = preg_match( '/<a\s/', $popup['body'] ) !== 0;
```
Do we actually need all of this logic? Would there be any harm in including all the triggers every time, and if the matching selector is absent it simply won't every trigger?

adekbadek replied:
Yes, errors are thrown in the JS console if an event handler is attached to a non-existent element :/

adekbadek commented:
1. Added the load event reporting: ab2d78b
2. Added the id to the label, so it's now `Title (id)` - might be the best of both worlds: ab2d78b
3. Committed your copy changes (thanks!): c754c1d6b2cec8650e33e5d1cd301a4282cff610

adekbadek commented:
- `Seen` issue should be fixed by 0cb1ce10254147968534c6bca553c6fb5e0de213
- `Load` issue should be fixed by 580d236e7297855da2ec823449dc2400d96e983e

adekbadek commented:
The Mailchimp form regex in this PR applies just to inline popups and specifically to the feature of not rendering the inline popup if the Mailchimp form was submitted. 
As for the analytics, both Mailchimp and Donations blocks render a `form` element. Submission of either will be reported as custom event, so this will be easy to track. I think we can assume that most of popups will render just one form, so it will be easy to discern those in Analytics (e.g. `Form Submission` event on popup with title `October Donations Campaign`).
